client/src/details/opportunitydetails.cpp

```diff
@@ -171,10 +173,19 @@ void OpportunityDetails::updateLinkedItemsButtons()
         mUi->viewNotesButton->setEnabled(!oppId.isEmpty());
     }
     {
-        const int documents = oppId.isEmpty() ? 0 : mLinkedItemsRepository->documentsForOpportunity(oppId).count();
-        const QString buttonText = (documents == 0) ? i18n("Attach Document") : i18np("View 1 Document", "View %1 Documents", documents);
-        mUi->viewDocumentsButton->setText(buttonText);
-        mUi->viewDocumentsButton->setEnabled(!oppId.isEmpty());
+        const auto documents = mLinkedItemsRepository->documentsForOpportunity(oppId);
+        const QString buttonText = documents.isEmpty() ? i18n("Attach Document") : i18np("Manage 1 Document", "Manage %1 Documents", documents.count());
+        mUi->manageDocumentsButton->setText(buttonText);
+        mUi->viewDocumentsButton->setEnabled(!documents.isEmpty());
+
+        auto *menu = new QMenu(this);
+        for (auto doc : documents) {
+            auto *action = menu->addAction(doc.documentName().toHtmlEscaped());
+            connect(action, &QAction::triggered, this, [this, doc]() {
+                ExternalOpen::openSugarDocument(doc.documentRevisionId(), resourceIdentifier(), this);
+            });
+        }
+        mUi->viewDocumentsButton->setMenu(menu);
```

Comment:
Technically you should delete the existing menu, otherwise they pile up (as children of "this"), if this method is called multiple times. Not a big deal in practice though, I guess one might see only a handful of updates of this opp or its documents while the dialog is up.